src/widget/wpixmapstore.cpp

```diff
@@ -13,16 +14,19 @@ PaintablePointer WPixmapStore::getPaintable(const PixmapSource& source,
     if (source.isEmpty()) {
         return PaintablePointer();
     }
-    QString key = QStringLiteral("%1%2%3").arg(source.getId(),
-            QString::number(mode),
-            QString::number(scaleFactor));
+    // Generate a key like: 'skins:LateNight/palemoon/buttons/btn___active.svg|STRETCH|1'
+    QString key = QString("%1|%2|%3")
+                          .arg(source.getPath(),
+                                  Paintable::DrawModeToString(mode),
+                                  QString::number(scaleFactor, 'g'));
```

Comment:
not a requirement for this PR, but a future optimization opportunity would be to avoid the QString entirely. Instead create a new struct that contains all the required data in raw binary form and then implement `qHash` (or whatever interface QHash needs for custom types). That would avoid the 4 temporary strings here. Does that make sense?

Reply:
As the path is already a string and the other components are tiny, I expect that this is already a cheap operation.

Reply:
Qt does no small-string-optimization, so it involves at least an atomic refcount increase, 1-2 string allocations as well as an additional one for the formatstring assembly. Then that string still needs to get hashed. Everything except the hashing could be avoided. If the path is not hot, then it probably doesn't make any difference, you're right in that regard. Still, this feels like a simple-enough optimization.

I will try to benchmark this in the future.